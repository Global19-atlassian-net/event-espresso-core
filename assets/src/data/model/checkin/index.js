export * from './constants';
export * from './query';
export * from './get-check-in-status-configuration';
