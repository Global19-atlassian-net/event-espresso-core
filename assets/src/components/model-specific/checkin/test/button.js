/**
 * External imports
 */
import { shallow } from 'enzyme';
import { CheckinFactory, AuthedCheckinResponse } from '@test/fixtures';
import { checkInModel } from '@eventespresso/model';

/**
 * Internal imports
 */
import CheckInButton from '../button';

const {
	STATUS_CHECKED_NEVER: neverChecked,
	STATUS_CHECKED_IN: checkIn,
	STATUS_CHECKED_OUT: checkOut,
} = checkInModel.CHECKIN_STATUS_ID;

describe( 'CheckInButton', () => {
	const getCheckin = ( status ) => {
		if ( status === neverChecked ) {
			return null;
		}
		const entity = CheckinFactory.fromExisting( AuthedCheckinResponse );
		entity.in = status;
		return entity;
	};
	const rendered = ( props ) => shallow( <CheckInButton { ...props } /> );
	it( 'renders as expected for default props', () => {
		const wrapper = rendered();
		expect( wrapper ).toMatchSnapshot();
	} );
	describe( 'renders expected classname for various checkin actions', () => {
		const description = 'should return the css class "$expectedClass" and the ' +
			'"$expectedText" for the given status of "$status" and with the ' +
			'force argument value of "$force"';
		it.each`
			status | expectedClass | expectedText | force
			${ neverChecked } | ${ 'ee-green' } | ${ 'Check In' } | ${ false }
			${ neverChecked } | ${ 'ee-yellow' } | ${ 'Check In Anyways' } | ${ true }
			${ checkIn } | ${ 'ee-red' } | ${ 'Check Out' } | ${ false }
			${ checkOut } | ${ 'ee-green' } | ${ 'Check In' } | ${ false }
		`( description,
	( { status, expectedClass, expectedText, force } ) => {
		const wrapper = rendered( {
			checkinEntity: getCheckin( status ),
			force,
		} );
		const button = wrapper.find( '.ee-button' );
		expect( button.hasClass( 'ee-roundish', expectedClass ) ).toBe( true );
		expect( button.text() )
			.toEqual( expect.stringContaining( expectedText ) );
	} );
	} );
} );
