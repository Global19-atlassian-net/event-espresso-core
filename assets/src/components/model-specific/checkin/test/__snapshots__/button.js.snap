// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckInButton renders as expected for default props 1`] = `
<ForwardRef(Button)
  className="ee-green ee-button ee-roundish"
  onClick={[Function]}
>
  Check In
</ForwardRef(Button)>
`;
